verilog/CST/expression.cc

```diff
@@ -72,6 +72,32 @@ bool ConstantIntegerValue(const verible::Symbol& expr, int* value) {
   return absl::SimpleAtoi(text, value);
 }
 
+
+const verible::Symbol* UnwrapExpression(const verible::Symbol& expr) {
+  
+  const auto& node = verible::SymbolCastToNode(expr);
+  const auto tag = static_cast<verilog::NodeEnum>(node.Tag().tag);
+
+  if (expr.Kind() == SymbolKind::kLeaf || tag != NodeEnum::kExpression) {
+    return &expr;
+  }
+
+  const auto& children = node.children();
+  return children.front().get();
+}
+
+const verible::Symbol* GetConditionExpressionPredicate(const verible::Symbol& condition_expr) {
+  return GetSubtreeAsSymbol(condition_expr, NodeEnum::kConditionExpression, 0);
+}
+
+const verible::Symbol* GetConditionExpressionTrueCase(const verible::Symbol& condition_expr) {
+  return GetSubtreeAsSymbol(condition_expr, NodeEnum::kConditionExpression, 2);
+}
+
+const verible::Symbol* GetConditionExpressionFalseCase(const verible::Symbol& condition_expr) {
+  return GetSubtreeAsSymbol(condition_expr, NodeEnum::kConditionExpression, 4);
+}
```

Comment:
Let's add unit tests for these, so that should any of these get restructured, we'll know to fix them in one place.

[Example unit tests](https://cs.opensource.google/verible/verible/+/master:verilog/CST/declaration_test.cc).
Or look at many other CST/*_test.cc files.

A typical recipe for unit testing these extractor functions:
1) FindAll of the outer structure of interest (might need to write a quick function for this too), in this case, condition expressions.
2) Write test cases, tagging the substring of text that corresponds to what an accessor function should return: condition, true case, false case.
3) Pass the array of test cases to `TestVerilogSyntaxRangeMatches` (this is where you call FindAll and possibly other functions to refine your search).